### ISSUE TITLE
False Positive when Firefox developer tools enabled

### ISSUE THREAD

**Issue by skyclad0x7b7** [fingerprintjs/BotD#157]
<img width="1728" alt="image" src="https://github.com/fingerprintjs/BotD/assets/13326895/2c33dfac-3dcc-4f0a-8d35-188d501e63b3">

<details>
    <summary>Logs</summary>

```
{"isError":false,"collectionTime":12,"detectionTime":13,"detectionResult":{"bot":true,"botKind":"unknown"},"detectedBot":"unknown","collectedData":{"rtt":{"state":-1,"error":"BotdError: navigator.connection is undefined"},"process":{"state":-1,"error":"BotdError: window.process is undefined"},"android":{"value":false,"state":0},"browserKind":{"value":"safari","state":0},"browserEngineKind":{"value":"gecko","state":0},"documentFocus":{"value":false,"state":0},"userAgent":{"value":"Mozilla/5.0 (Linux; Android 11; SAMSUNG SM-G973U) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/14.2 Chrome/87.0.4280.141 Mobile Safari/537.36","state":0},"appVersion":{"value":"5.0 (Macintosh)","state":0},"windowSize":{"value":{"outerWidth":412,"outerHeight":915,"innerWidth":412,"innerHeight":915},"state":0},"pluginsLength":{"value":5,"state":0},"pluginsArray":{"value":true,"state":0},"errorTrace":{"value":"errorTrace@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:28945\ny/</</</</<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:22812\no/c/</<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:21466\no/c/<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:21571\ne/<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:20487\ne@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:20232\ny/</</<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:22649\ny/</<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:22625\no/c/</<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:21466\no/c/<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:21571\ne/<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:20487\ne@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:20232\ny@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:22481\nn.prototype.collect/D</<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:34172\no/c/</<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:21466\no/c/<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:21571\ne/<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:20487\ne@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:20232\nn.prototype.collect@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:34066\nY/</<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:34646\no/c/</<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:21466\no/c/<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:21571\ne/<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:20487\ne@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:20232\nY@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:34329\ntn/</</</<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:35274\no/c/</<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:21466\no/c/<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:21571\ne/<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:20487\ne@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:20232\ntn/</<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:35113\no/c/</<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:21466\no/c/<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:21571\ne/<@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:20487\ne@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:20232\ntn@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:35013\nwindow.onload@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:37547\nEventHandlerNonNull*@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:37231\n@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:37554\n@https://fingerprintjs.github.io/BotD/main/main.bundle.js:1:37558\n","state":0},"productSub":{"value":"20100101","state":0},"windowExternal":{"value":"[object External]","state":0},"mimeTypesConsistent":{"value":true,"state":0},"evalLength":{"value":37,"state":0},"webGL":{"value":{"vendor":"Mozilla","renderer":"Apple M1"},"state":0},"webDriver":{"value":false,"state":0},"languages":{"value":[["ko-KR"],["ko-KR","ko","en-US","en"]],"state":0},"documentElementKeys":{"value":["lang"],"state":0},"functionBind":{"value":"function bind() {\n    [native code]\n}","state":0},"distinctiveProps":{"value":{"awesomium":false,"cef":false,"cefsharp":false,"coachjs":false,"fminer":false,"geb":false,"nightmarejs":false,"phantomas":false,"phantomjs":false,"rhino":false,"selenium":false,"webdriverio":false,"webdriver":false,"headless_chrome":false},"state":0},"notificationPermissions":{"value":false,"state":0}},"detectorsResults":{"detectAppVersion":{"bot":false},"detectDocumentAttributes":{"bot":false},"detectErrorTrace":{"bot":false},"detectEvalLengthInconsistency":{"bot":false},"detectFunctionBind":{"bot":false},"detectLanguagesLengthInconsistency":{"bot":false},"detectNotificationPermissions":{"bot":false},"detectPluginsArray":{"bot":false},"detectPluginsLengthInconsistency":{"bot":false},"detectProcess":{"bot":false},"detectUserAgent":{"bot":false},"detectWebDriver":{"bot":false},"detectWebGL":{"bot":false},"detectWindowExternal":{"bot":false},"detectWindowSize":{"bot":false},"detectMimeTypesConsistent":{"bot":false},"detectProductSub":{"bot":true,"botKind":"unknown"},"detectDistinctiveProperties":{"bot":false}},"debugData":{"browserEngineKind":"gecko","browserKind":"safari","documentFocus":false,"mozAppearanceSupport":true,"isAndroid":false,"isDesktopWebKit":false}}
```
</details>

System: Apple M1 Pro, macOS Sonoma 14.1.1
Browser: Firefox Browser 120.0.1 (64-bit)

When Firefox developer tools and Responsive Design Mode enabled, Botd detect it as a bot.

**Comment by xnerhu** [comment]
I can confirm it also detected unknown on windows

**Comment by xnerhu** [comment]
Apart from a bug, which I just fixed, where `getBrowserKind` returned safari instead of chrome. This detection seems to be justified.

Firefox in this case seems to replace user agent with (at least on my side) `Mozilla/5.0 (Linux; Android 11; SAMSUNG SM-G973U) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/14.2 Chrome/87.0.4280.141 Mobile Safari/537.36`, which looks like chrome. Then `detectProductSub` detector looks that `navigator.productSub` and finds mismatched value.

In short, you got it marked as bot, because firefox modified the user agent.
